gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -899,6 +914,63 @@ public SubscriptionEntity close(final ExecutionContext executionContext, String
         }
     }
 
+    @Override
+    public SubscriptionEntity pauseByConsumer(ExecutionContext executionContext, String subscriptionId) {
+        try {
+            logger.debug("Pause subscription {} by consumer", subscriptionId);
+
+            Subscription subscription = subscriptionRepository
+                .findById(subscriptionId)
+                .orElseThrow(() -> new SubscriptionNotFoundException(subscriptionId));
+
+            final ApplicationEntity application = applicationService.findById(executionContext, subscription.getApplication());
+            final GenericPlanEntity genericPlanEntity = planSearchService.findById(executionContext, subscription.getPlan());
+            String apiId = genericPlanEntity.getApiId();
+            final GenericApiModel genericApiModel = apiTemplateService.findByIdForTemplates(executionContext, apiId);
+            validateConsumerStatus(subscription, genericApiModel);
+
+            // Here, do not care about the status managed by the publisher. The subscription will be active on the gateway if consumerStatus == ACTIVE and status == ACCEPTED
+            if (subscription.getConsumerStatus() == Subscription.ConsumerStatus.ACTIVE) {
+                Subscription previousSubscription = new Subscription(subscription);
+                final Date now = new Date();
+                subscription.setUpdatedAt(now);
+                subscription.setConsumerPausedAt(now);
+                subscription.setConsumerStatus(Subscription.ConsumerStatus.PAUSED);
+
+                subscription = subscriptionRepository.update(subscription);
+
+                createAudit(
+                    executionContext,
+                    apiId,
+                    subscription.getApplication(),
+                    SUBSCRIPTION_PAUSED_BY_CONSUMER,
+                    subscription.getUpdatedAt(),
+                    previousSubscription,
+                    subscription
+                );
+
+                // active API Keys are automatically paused (if they are not shared)
+                if (!application.hasApiKeySharedMode()) {
+                    streamActiveApiKeys(executionContext, subscription.getId())
+                        .forEach(
+                            apiKey -> {
+                                apiKey.setPaused(true);
+                                apiKeyService.update(executionContext, apiKey);
+                            }
+                        );
+                }
```

Comment:
Duplicated from `pause()` method and could be mutualized.

Reply:
exact